### PR TITLE
Feature/activity

### PR DESCRIPTION
**Pull request by AhmedMahdy14**

### CODE REVIEW

File: apihub_users/common/db_session.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ def get_db_engine():
     )
 
 
+DB_ENGINE = get_db_engine()
```
Comment (yifan) [inline apihub_users/common/db_session.py:27]: not used

File: apihub_users/usage/schemas.py [hunk 2]
```diff
@@ -16,3 +18,28 @@ class UsageCreate(UsageBase):
...
 class UsageDetails(UsageCreate):
     pass
+
+
+class ActivityBase(BaseModel):
+    pass
+
+
+class ActivityCreate(ActivityBase):
+    request: Optional[str] = None
```
Comment (yifan) [inline apihub_users/usage/schemas.py:28]: is Optional necessary? Let's think about how we are going to use this.

File: apihub_users/usage/schemas.py [hunk 3]
```diff
@@ -16,3 +18,28 @@ class UsageCreate(UsageBase):
...
+    username: Optional[str] = None
+    tier: Optional[str] = None
+    status: Optional[str] = None
+    request_key: Optional[str] = None
+    result: Optional[str] = None
```
Comment (yifan) [inline apihub_users/usage/schemas.py:33]: Can result be a BaseModel?

File: apihub_users/usage/schemas.py [hunk 4]
```diff
@@ -16,3 +18,28 @@ class UsageCreate(UsageBase):
...
+    payload: Optional[str] = None
```
Comment (yifan) [inline apihub_users/usage/schemas.py:34]: what if we make this a JSON or something?

**Review by yifan — COMMENTED** [review]
minor changes

File: apihub_users/usage/schemas.py (hunk 2 above)
Comment (AhmedMahdy14) [reply apihub_users/usage/schemas.py:28]: I think it should be mandatory.

File: apihub_users/common/db_session.py (hunk 1 above)
Comment (AhmedMahdy14) [reply apihub_users/common/db_session.py:27]: it's been used at admin.

File: apihub_users/usage/schemas.py (hunk 3 above)
Comment (AhmedMahdy14) [reply apihub_users/usage/schemas.py:33]: yes I think so.

File: apihub_users/usage/schemas.py (hunk 4 above)
Comment (AhmedMahdy14) [reply apihub_users/usage/schemas.py:34]: you mean result or these four fields ?

File: apihub_users/usage/schemas.py (hunk 3 above)
Comment (AhmedMahdy14) [reply apihub_users/usage/schemas.py:33]: I will keep result as string, we can dump the actual result form Result.py into stringified dictionary. 
I tried to make it as model it complicated things and I don't see it's worthy the effort.

File: apihub_users/subscription/depends.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 from fastapi_jwt_auth import AuthJWT
 from redis import Redis
 
+from apihub_users.subscription.schemas import SubscriptionBase
```
Comment (yifan) [inline apihub_users/subscription/depends.py:5]: please use relative import here

File: apihub_users/subscription/router.py [hunk 6]
```diff
@@ -83,11 +87,14 @@ def get_active_subscription(
     user: UserBase = Depends(require_token),
     session=Depends(create_session),
 ):
+
     query = SubscriptionQuery(session)
     try:
         subscription = query.get_active_subscription(user.username, application)
     except SubscriptionException:
-        return {}
+        return JSONResponse(
```
Comment (yifan) [inline apihub_users/subscription/router.py:95]: use HTTPException

File: apihub_users/subscription/router.py [hunk 7]
```diff
@@ -99,6 +106,8 @@ def get_active_subscriptions(
 ):
     if user.is_user:
```
Comment (yifan) [inline apihub_users/subscription/router.py:107]:
```
if not user.is_user:
    return []
....
return SubscriptionTokenResponse(
        username=user.username,
```

**Review by yifan — COMMENTED** [review]
minor comments

File: apihub_users/usage/schemas.py (hunk 4 above)
Comment (yifan) [reply apihub_users/usage/schemas.py]: I mean payload, but it is fine you keep it simple as str

**Review by yifan — APPROVED** [review]

**Comment by yifan** [comment]
looks good

**Pull request merged**